lib/gutentag/tagged_with/query.rb

```diff
@@ -20,7 +21,7 @@ def model_id
   end
 
   def query
-    return taggable_ids_query if match == :any || values.length == 1
+    return taggable_ids_query if %i[any none].include?(match) || values.length == 1
```

Comment:
And the condition here (both parts) can probably be shifted into a private method too… maybe `match_any_or_none?`. Even prior to your change, it took me a moment to realise the purpose of the condition, so a clear private method name hopefully improves clarity.